File: mach/src/platform/plugins/transformer_noop/transformer.rs
use crate::public::Config;
use crate::public::MutableAsset;
use crate::public::Transformer;

#[derive(Debug)]
pub struct DefaultTransformerNoop {}

impl Transformer for DefaultTransformerNoop {
  fn transform(
    &self,
    _asset: &mut MutableAsset,
    _config: &Config,
  ) -> Result<(), String> {
    return Ok(());
  }
}

File: testing/fixtures/plugins-nodejs-transformer-txt/plugins/transformer.mjs
import { Transformer } from '@alshdavid/mach'

export default new Transformer({
  loadConfig() {
    return 'TextTransformerConfig'
  },
  async transform({ asset, config }) {
    // Testing if loadConfig works
    if (typeof config !== 'string' || config !== 'TextTransformerConfig') {
      throw new Error('Did not load config')
    }

    const contents = await asset.getCode()

    asset.setCode(`
      export default "${contents}"
    `)

    asset.type = 'js'

    return asset
  },
})
